agent/src/main/java/com/appland/appmap/output/v1/Event.java

```diff
@@ -76,6 +83,11 @@ public class Event {
 
   private LocalDateTime startTime;
 
+  // If pending is true, this event needs to be updated before it can be written
+  // to the AppMap. 
+  private boolean isPending = false;
```

Comment:
Can you add some more comments about how isPending and pendingEvent works, and why it's necessary? Because I am not figuring it out as easily as I'd like to :-)

Reply:
Before these changes, adding an event to the session resulted in it being serialized and written to the temporary file. This meant that if the event was updated subsequently, those changes would be ignored.

For example, Spring doesn't set `HandlerMapping.BEST_MATCHING_PATTERN_ATTRIBUTE` and `HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE` until after the response has been generated. So, we need to save the `http_server_request` event, then set `normalized_path` and the `message` parameters when processing the response.

If that makes it clearer, I'll add comments to that effect.

Reply:
Yes, that comment would be very helpful.

Reply:
Done.